core/parachain/approval/approval_distribution.cpp

```diff
@@ -1511,7 +1775,34 @@ namespace kagome::parachain {
                      })
             .first->second;
 
-    runDistributeAssignment(assignment, claimed_candidate_index);
+    auto peer_filter = [&](const auto &peer, const auto &peer_kn) {
+      if (source && peer == source->get()) {
+        return false;
+      }
+
+      const bool already_sent =
+          peer_kn.sent.contains(message_subject,
+                                approval::MessageKindAssignment{})
+          || peer_kn.sent.contains(message_subject,
+                                   approval::MessageKindApproval{});
+      return !already_sent;
+    };
+
+    std::unordered_set<libp2p::peer::PeerId> peers{};
+    for (const auto &[peer_id, peer_knowledge] : entry.known_by) {
+      if (peer_filter(peer_id, peer_knowledge)) {
+        peers.insert(peer_id);
+        if (auto it = entry.known_by.find(peer_id);
+            it != entry.known_by.end()) {
```

Comment:
This condition looks degenerate (always `true` because `peer_id` is gotten in `for`)